src/libcimplot.jl

```diff
@@ -0,0 +1,1583 @@
+end
+
+function ImPlotRange_Contains(self, value)
+    ccall((:ImPlotRange_Contains, libcimplot), Bool, (Ptr{ImPlotRange}, Cdouble), self, value)
+end
+
+function ImPlotRange_Size(self)
+    ccall((:ImPlotRange_Size, libcimplot), Cdouble, (Ptr{ImPlotRange},), self)
+end
+
+function ImPlotLimits_ContainsPlotPoInt(self, p)
+    ccall((:ImPlotLimits_ContainsPlotPoInt, libcimplot), Bool, (Ptr{ImPlotLimits}, ImPlotPoint), self, p)
+end
+
+function ImPlotLimits_Containsdouble(self, x, y)
+    ccall((:ImPlotLimits_Containsdouble, libcimplot), Bool, (Ptr{ImPlotLimits}, Cdouble, Cdouble), self, x, y)
+end
+
+function ImPlotStyle_ImPlotStyle()
+    ccall((:ImPlotStyle_ImPlotStyle, libcimplot), Ptr{ImPlotStyle}, ())
+end
+
+function ImPlotStyle_destroy(self)
+    ccall((:ImPlotStyle_destroy, libcimplot), Cvoid, (Ptr{ImPlotStyle},), self)
+end
+
+function ImPlotInputMap_ImPlotInputMap()
+    ccall((:ImPlotInputMap_ImPlotInputMap, libcimplot), Ptr{ImPlotInputMap}, ())
+end
+
+function ImPlotInputMap_destroy(self)
+    ccall((:ImPlotInputMap_destroy, libcimplot), Cvoid, (Ptr{ImPlotInputMap},), self)
+end
+
+function CreateContext()
+    ccall((:ImPlot_CreateContext, libcimplot), Ptr{ImPlotContext}, ())
+end
+
+function DestroyContext(ctx)
+    ccall((:ImPlot_DestroyContext, libcimplot), Cvoid, (Ptr{ImPlotContext},), ctx)
+end
+
+function GetCurrentContext()
+    ccall((:ImPlot_GetCurrentContext, libcimplot), Ptr{ImPlotContext}, ())
+end
+
+function SetCurrentContext(ctx)
+    ccall((:ImPlot_SetCurrentContext, libcimplot), Cvoid, (Ptr{ImPlotContext},), ctx)
+end
+
+function BeginPlot(title_id, x_label, y_label, size, flags, x_flags, y_flags, y2_flags, y3_flags)
+    ccall((:ImPlot_BeginPlot, libcimplot), Bool, (Cstring, Cstring, Cstring, ImVec2, ImPlotFlags, ImPlotAxisFlags, ImPlotAxisFlags, ImPlotAxisFlags, ImPlotAxisFlags), title_id, x_label, y_label, size, flags, x_flags, y_flags, y2_flags, y3_flags)
+end
+
+function EndPlot()
+    ccall((:ImPlot_EndPlot, libcimplot), Cvoid, ())
+end
+
+function PlotLine(label_id, values::AbstractArray{Cfloat}, count::Integer, xscale::Real, x0::Real, offset::Integer, stride::Integer)
+    ccall((:ImPlot_PlotLineFloatPtrInt, libcimplot), Cvoid, (Cstring, Ref{Cfloat}, Cint, Cdouble, Cdouble, Cint, Cint), label_id, values, count, xscale, x0, offset, stride)
+end
+
+function PlotLine(label_id, values::AbstractArray{Cdouble}, count::Integer, xscale::Real, x0::Real, offset::Integer, stride::Integer)
+    ccall((:ImPlot_PlotLinedoublePtrInt, libcimplot), Cvoid, (Cstring, Ref{Cdouble}, Cint, Cdouble, Cdouble, Cint, Cint), label_id, values, count, xscale, x0, offset, stride)
+end
+
+function PlotLine(label_id, values::AbstractArray{ImS8}, count::Integer, xscale::Real, x0::Real, offset::Integer, stride::Integer)
+    ccall((:ImPlot_PlotLineS8PtrInt, libcimplot), Cvoid, (Cstring, Ref{ImS8}, Cint, Cdouble, Cdouble, Cint, Cint), label_id, values, count, xscale, x0, offset, stride)
+end
```

Comment:
@Gnimuc @sairus7 @yamen I wrote an extra pass for all PlotXXX functions. They are now typed loosely so we can let multiple dispatch do the work for us. Still need to clean up the rest, but the idea should be clear. I would plan to rebase the changes happening in #14 onto the updated wrappers here, thereby making any kind of symbol lookup unnecessary. It should also make it easier to finish writing the demo.

Reply:
Are there any default value handling for `count`, `xscale`, `x0`, `offset`, `stride`? Seems like it remains the only reason to have helper functions.

Reply:
It's possible--I even left some commented out sections in the generator script that provide examples of how to do it. However, I'm not sure whether it's worth it to take it that far. I may consider taking it an extra step in the future. For now, this change (along with some auto exporting Clang can now include) will reduce the vast majority of the work anyhow. (e.g. since everything is overloaded, adding defaults to match the behavior of C++ is just a 1-2 line wrapper)

Reply:
Really nice! Agree it may not be immediately worth adding defaults and other conveniences into the wrapper scripts as there may be nicer / cleaner ways of managing that in Julia directly.